### PR TITLE
feat(core): add daily count for journal sidebar

### PR DESCRIPTION
**Pull request by CatsJuice**

### CODE REVIEW

**Comment by CatsJuice** [comment]
Current dependencies on/for this PR:
* `canary`
  * **PR #5468** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5468?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #5477** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5477?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #5517** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5517?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #5549** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5549?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #5557** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5557?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #5558** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5558?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #5559** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5559?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
                * **PR #5574** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5574?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                  * **PR #5578** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5578?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                    * **PR #5579** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5579?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/toeverything/AFFiNE/5559?utm_source=stack-comment).

File: packages/frontend/core/src/pages/workspace/detail-page/editor-sidebar/extensions/journal.tsx [hunk 1]
```diff
@@ -28,12 +78,148 @@ const EditorJournalPanel = ({ workspace, page }: EditorExtensionProps) => {
...
+    </div>
+  );
+};
+
+const sortPagesByDate = (
+  pages: Page[],
+  field: 'updatedDate' | 'createDate',
+  order: 'asc' | 'desc' = 'desc'
+) => {
+  return pages.sort((a, b) => {
```
Comment (pengx17) [inline packages/frontend/core/src/pages/workspace/detail-page/editor-sidebar/extensions/journal.tsx]: better not sort in place. change to `[...pages].sort` intead

File: packages/frontend/core/src/pages/workspace/detail-page/editor-sidebar/extensions/journal.tsx [hunk 2]
```diff
@@ -28,12 +78,148 @@ const EditorJournalPanel = ({ workspace, page }: EditorExtensionProps) => {
...
+
+  useEffect(() => {
+    const created: Page[] = [];
+    const updated: Page[] = [];
+    const today = journalDate;
+    Array.from(workspace.pages.values()).forEach(page => {
+      if (page.meta.trash) return;
+      if (
+        page.meta.createDate &&
+        dayjs(page.meta.createDate).isSame(today, 'day')
```
Comment (pengx17) [inline packages/frontend/core/src/pages/workspace/detail-page/editor-sidebar/extensions/journal.tsx]: maybe a page is both created and updated on the same day?

File: packages/frontend/core/src/pages/workspace/detail-page/editor-sidebar/extensions/journal.tsx [hunk 3]
```diff
@@ -28,12 +78,148 @@ const EditorJournalPanel = ({ workspace, page }: EditorExtensionProps) => {
...
+    });
+    setCreatedToday(sortPagesByDate(created, 'createDate'));
+    setUpdatedToday(sortPagesByDate(updated, 'updatedDate'));
+  }, [journalDate, workspace.pages]);
+
+  const activeIndex = headerItems.findIndex(({ name }) => name === activeItem);
+  const vars = {
+    '--active-index': activeIndex,
+    '--item-count': headerItems.length,
+  } as CSSProperties;
```
Comment (pengx17) [inline packages/frontend/core/src/pages/workspace/detail-page/editor-sidebar/extensions/journal.tsx]: may use `assignInlineVars` from `@vanilla-extract/dynamic`

File: packages/frontend/core/src/pages/workspace/detail-page/editor-sidebar/extensions/journal.tsx [hunk 4]
```diff
@@ -28,12 +79,150 @@ const EditorJournalPanel = ({ workspace, page }: EditorExtensionProps) => {
...
+      }
+      if (
+        page.meta.updatedDate &&
+        dayjs(page.meta.updatedDate).isSame(today, 'day')
+      ) {
+        updated.push(page);
+      }
+    });
+    setCreatedToday(sortPagesByDate(created, 'createDate'));
+    setUpdatedToday(sortPagesByDate(updated, 'updatedDate'));
```
Comment (joooye34) [inline packages/frontend/core/src/pages/workspace/detail-page/editor-sidebar/extensions/journal.tsx]: Seems the created list and updated list should be calculation memos, not a state.

File: packages/frontend/core/src/pages/workspace/detail-page/editor-sidebar/extensions/journal.tsx [hunk 5]
```diff
@@ -28,12 +79,151 @@ const EditorJournalPanel = ({ workspace, page }: EditorExtensionProps) => {
...
+};
+const JournalDailyCountBlock = ({ workspace, page }: EditorExtensionProps) => {
+  const nodeRef = useRef<HTMLDivElement>(null);
+  const t = useAFFiNEI18N();
+  const { journalDate } = useJournalInfoHelper(page?.meta);
+  const [activeItem, setActiveItem] = useState<NavItemName>('createdToday');
+
+  const navigateHelper = useNavigateHelper();
+
+  const getTodaysPages = useCallback(
```
Comment (pengx17) [inline packages/frontend/core/src/pages/workspace/detail-page/editor-sidebar/extensions/journal.tsx:127]: Why using a callback? I think it is better to return the calculated value

**Review by pengx17 — APPROVED** [review]

**Pull request merged**